src/main/java/org/icgc/argo/program_service/model/entity/Cancer.java

```diff
@@ -0,0 +1,45 @@
+package org.icgc.argo.program_service.model.entity;
+
+import lombok.*;
+import org.hibernate.annotations.GenericGenerator;
+import org.icgc.argo.program_service.model.enums.JavaFields;
+import org.icgc.argo.program_service.model.enums.SqlFields;
+import org.icgc.argo.program_service.model.enums.Tables;
+import org.icgc.argo.program_service.model.join.ProgramCancer;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.google.common.collect.Sets.newHashSet;
+
+
+@Entity
+@Table(name = Tables.CANCER)
+@Builder
+@Data
+@EqualsAndHashCode(of = "name")
+@AllArgsConstructor
+@NoArgsConstructor
+public class Cancer implements NameableEntity<UUID> {
```

Comment:
Please add the following new lombok functionality [@FieldNameConstants](https://projectlombok.org/features/experimental/FieldNameConstants). This will convieniently generate a `Cancer.Fields` public static inner class, which can be used for when doing dynamic joins or when we need to specify a field name

Reply:
This will also eliminate the need for the `JavaFields` class!!

Reply:
FieldNameConstants has an issue with intellij lombok plugin version < 0.24, when referring to annotated inner class ```Field```, the field is showing ```myClass.Field.NULLmyFieldNULL```,
see this: https://github.com/mplushnikov/lombok-intellij-plugin/issues/547, updating plugin will solve this issue, please make sure the lombok plugin is the latest 0.24 version.